create-plugin/src/test/resources/sources/HeaderParameters.kt

```diff
@@ -0,0 +1,92 @@
+package sources
+
+import io.github.tabilzad.ktor.annotations.GenerateOpenApi
+import io.ktor.server.application.*
+import io.ktor.server.routing.*
+import sources.MyStringQueries.queryParam
+
+enum class MyQueries(val param: String) {
+    ORDER_ID("orderId")
+}
+
+object MyStringQueries {
+    val queryParam = "my_string_query_param"
+}
+
+@GenerateOpenApi
+fun Application.queryParametersTest() {
+    routing {
+        route("/v1") {
+            get("/order10") {
+                val query = call.request.queryParameters["my_query_param"]
+                val header = call.request.headers["X_API_KEY"]
```

Comment:
Ah yes makes sense, let me look into that. Will update the tests

Reply:
I added some more test cases to get the other scenarios covered. For the default headers i gave it a try but moved it to another pr since i could not get it going easily